pkg/cluster/delete_test.go

```diff
@@ -49,25 +53,25 @@ func TestDeleteNic(t *testing.T) {
 			name: "nic is in succeeded provisioning state",
 			mocks: func(networkInterfaces *mock_network.MockInterfacesClient) {
 				nic.InterfacePropertiesFormat.ProvisioningState = mgmtnetwork.Succeeded
-				networkInterfaces.EXPECT().Get(gomock.Any(), clusterRG, nicName, "").Return(nic, nil)
-				networkInterfaces.EXPECT().DeleteAndWait(gomock.Any(), clusterRG, nicName).Return(nil)
+				networkInterfaces.EXPECT().Get(deprecratedgomock.Any(), clusterRG, nicName, "").Return(nic, nil)
```

Comment:
Wait, never mind... I tried again and I was reminded of why I didn't do it in the first place: these other mock clients need the `deprecatedgomock` because they are generated by us, and they currently use the deprecated one instead of the Uber one.

I agree that we should switch to the Uber module, but I think changing the way the mocks are generated is out of scope for this PR.